File: 0447-Number-of-Boomerangs/0447.py
class Solution:
    def numberOfBoomerangs(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        def dis(point1, point2):
            return (point1[0] - point2[0])**2 + (point1[1] - point2[1])**2

        result = 0
        for i in points:
            record = {}
            for j in points:
                if j != i:
                    distance = dis(i, j)
                    record[distance] = record.get(distance, 0) + 1

            for _, val in record.items():
                if val >= 2:
                    result += (val - 1)* val

        return result

if __name__ == "__main__":
    points = [[0,0],[1,0],[-1,0],[0,1],[0,-1]]
    print(Solution().numberOfBoomerangs(points))

File: src/0447-Number-of-Boomerangs/0447.cpp
#include <iostream>
#include <vector>
#include <math.h>
#include <unordered_map>
using namespace std;
static int x = []() {std::ios::sync_with_stdio(false); cin.tie(0); return 0; }();
class Solution 
{
public:    
    int numberOfBoomerangs(vector<pair<int, int>>& points) 
    {
        int result = 0;
        for (auto& i : points)
        {
            unordered_map<int, int> record;
            for (auto& j : points)
            {
                if (j != i)
                {
                    ++record[dis(i, j)];
                }
            }
            for (auto& val : record)
            {
                if (val.second >= 2)
                {
                    result += val.second*(val.second - 1);
                }
            }
        }
        return result;
    }
private:
    static inline int dis(pair<int, int>& point1, pair<int, int>& point2)
    {
        return pow((point1.first - point2.first), 2) + pow((point1.second - point2.second), 2);
    }
};
int main()
{
    vector<pair<int, int>> points = {make_pair(0,0),make_pair(1,0),make_pair(2,0)};
    cout << Solution().numberOfBoomerangs(points);
    return 0;
}